File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    var answer = [];
    let k_dict = {}
    
    for(var i=0; i<keymap.length; i++){
        for(var j=0; j<keymap[i].length; j++){
            if(Object.keys(k_dict).includes(keymap[i][j])){
                if(k_dict[keymap[i][j]] > j+1){
                    k_dict[keymap[i][j]] = j+1
                } else {
                    continue
                }
            } else {
                k_dict[keymap[i][j]] = j+1
            }
        }
    }
    
    for(var k=0; k<targets.length; k++){
        let result = 0
        for(var l=0; l<targets[k].length; l++){
            if(!Object.keys(k_dict).includes(targets[k][l])){
                result = -1
                break
            } else {
                result += k_dict[targets[k][l]]
            }
        }
        answer.push(result)
    }
    return answer;
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    k_dict = {}
    
    for key in keymap:
        for k in range(len(key)):
            if key[k] in k_dict:
                k_dict[key[k]] = min(k+1,k_dict[key[k]])
            else:
                k_dict[key[k]] = k+1
    
    for target in targets:
        result = 0
        for t in target:
            if t in k_dict:
                result += k_dict[t]
            else:
                result = -1
                break
        answer.append(result)
            
    return answer